packages/flutter/lib/src/material/switch_list_tile.dart

```diff
@@ -187,19 +229,41 @@ class SwitchListTile extends StatelessWidget {
   /// Normally, this property is left to its default value, false.
   final bool selected;
 
+  /// If adaptive, creates the switch with [Switch.adaptive].
+  final _SwitchListTileType _switchListTileType;
+
   @override
   Widget build(BuildContext context) {
-    final Widget control = Switch(
-      value: value,
-      onChanged: onChanged,
-      activeColor: activeColor,
-      activeThumbImage: activeThumbImage,
-      inactiveThumbImage: inactiveThumbImage,
-      materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
-      activeTrackColor: activeTrackColor,
-      inactiveTrackColor: inactiveTrackColor,
-      inactiveThumbColor: inactiveThumbColor,
-    );
+    Widget control;
+    switch (_switchListTileType) {
+      case _SwitchListTileType.adaptive:
+        control = Switch.adaptive(
+          value: value,
+          onChanged: onChanged,
+          activeColor: activeColor,
+          activeThumbImage: activeThumbImage,
+          inactiveThumbImage: inactiveThumbImage,
+          materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          activeTrackColor: activeTrackColor,
+          inactiveTrackColor: inactiveTrackColor,
+          inactiveThumbColor: inactiveThumbColor,
+        );
+        break;
+
+      case _SwitchListTileType.material:
+      default:
```

Comment:
Don't include the `default` case. If another _SwitchListTileType enum value was added, we want the compiler to fail here.

Reply:
Good call.